File: bms/go/go-gt/heap_swap/heap_swap.go
package test

var ()

const ()

func (h Heap) swap(a, b int64) {
	i, j := h.I[a], h.I[b]
	h.I[a], h.I[b] = h.I[b], h.I[a]
	h.A[i], h.A[j] = b, a
}

type Heap struct {
	I []int64
	A []int64
}

File: translations_only/success/go/go-gt/heap_swap/gemini/BaseRepair/heap_swap.rs

fn swap(h: &mut Heap, a: i64, b: i64) {
    let i = h.I[a as usize];
    let j = h.I[b as usize];
    h.I[a as usize] = h.I[b as usize];
    h.I[b as usize] = i;
    h.A[i as usize] = b;
    h.A[j as usize] = a;
}

struct Heap {
    I: Vec<i64>,
    A: Vec<i64>,
}
